addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -136,10 +137,30 @@ def _create_invoice(self, order, so_line, amount):
                     subtype_id=self.env.ref('mail.mt_note').id)
         return invoice
 
+    def _downpayment_section_exists(self, order):
+        return len(
+            order.order_line.filtered(lambda line: line.display_type and line.is_downpayment)
+            ) >= 1
```

Comment:
No need for a separate method.
Also, you could use `any(line.is_downpayment and line.display_type for line in order.order_line)` instead ;)